java/arcs/core/data/expression/Builders.kt

```diff
@@ -121,13 +125,38 @@ infix fun Expression<out Any>.neq(other: Expression<out Any>) = Expression.Binar
 
 /** Constructs a [Expression.FieldExpression] given a [Expression.Scope] and [field]. */
 operator fun <E : Expression.Scope, T> E.get(field: String) = Expression.FieldExpression<E, T>(
-    Expression.ObjectLiteralExpression(this), field)
+    when (this) {
+        is CurrentScope<*> -> Expression.CurrentScopeExpression()
+        is MapScope<*> -> Expression.ObjectLiteralExpression(this as E)
```

Comment:
I am wondering do we need 2 different classes for CurrentScope and MapScope? Could we just have a single Scope?

Reply:
+1 I would like some more info on the difference between them too

Reply:
MapScope was used purely for testing previously. I may be able to remove it (and ObjectLiteralExpression)